src/EbmlSInteger.cpp

```diff
@@ -134,19 +134,24 @@ uint64 EbmlSInteger::UpdateSize(bool bWithDefault, bool /* bForceRender */)
 filepos_t EbmlSInteger::ReadData(IOCallback & input, ScopeMode ReadFully)
 {
   if (ReadFully != SCOPE_NO_DATA) {
-    binary Buffer[8];
-    input.readFully(Buffer, GetSize());
+    if (GetSize() > 8) {
+      // impossible to read, skip it
+      input.setFilePointer(GetSize(), seek_current);
+    } else {
```

Comment:
Again, I'd prefer to short-circuit here. Just `return GetSize();` after `input.setFilePointer`, remove the `else` and you'll save us a level of indentation. Fewer levels of indentation pretty much always make the code easier to follow & understand.

As a matter of fact, it'd be even better to short-circuit on the `SCOPE_NO_DATA` check as well:

```
  if (ReadFully == SCOPE_NO_DATA)
    return GetSize();

  if (GetSize() > 8) {
    // impossible to read, skip it
    input.setFilePointer(GetSize(), seek_current);
   return GetSize();
  }

  binary Buffer[8];
  …
```

That would save _two_ levels of indentation & the associated cost of remembering which closing parenthesis belongs to which conditional.

Reply:
* the less changes in a patch the easier it is to review
* return is usually better one one location for easier debugging or follow the flow of the code

But ok, I'll do it.